Orange/preprocess/preprocess.py

```diff
@@ -154,13 +155,21 @@ def __call__(self, data):
         if not self.force and not np.isnan(data.X).any():
             return data
         self.imputer = skl_preprocessing.Imputer(strategy=self.strategy)
-        X = self.imputer.fit_transform(data.X)
+        self.imputer.fit(data.X)
+        self.fix_statistics_for_discrete_vars(data)
```

Comment:
This class is only a wrapper for the scikit imputer and I don't like "correcting" the imputed values based on var types. The `strategy` parameter is passed to scikit and the result should be the same as in scikit.

If we want something more useful for Orange, I suggest we add an additional 'smart' strategy (can even be default) which uses 'mean' on continuous vars and 'most_frequent' on discrete.